Cargo.toml

```diff
@@ -90,15 +90,15 @@ arrow = { version = "54.2.1", features = [
     "prettyprint",
     "chrono-tz",
 ] }
-arrow-buffer = { version = "54.1.0", default-features = false }
+arrow-buffer = { version = "54.3.0", default-features = false }
 arrow-flight = { version = "54.2.1", features = [
```

Comment:
Why do we reference various arrow deps with different versions (54.1.0, 54.2.0, 54.2.1) in this file?

Reply:
I think it is a function of whatever dependabot is doing

in general we had been keeping this at the latest released version, but there has been some discussion 

- Related discussion https://github.com/apache/datafusion/issues/14962